src/input/pointer.js

```diff
@@ -241,68 +237,141 @@
+										for (i = handlers.callbacks[activeEventList[POINTER_LEAVE]].length - 1; (callback = handlers.callbacks[activeEventList[POINTER_LEAVE]][i]); i--) {
+											// delete the pointerId
+											handlers.pointerId = null;
+											if (callback(e) === false) {
+												// stop propagating the event if return false
+												handled = true;
+												break;
+											}
+										}
+									}
+								}
+							} else if (handlers.pointerId === null && eventInBounds) {
+								// no pointer & moved inside of bounds: trigger the POINTER_ENTER callbacks instead
+								if (handlers.callbacks[activeEventList[POINTER_ENTER]]) {
+									for (i = handlers.callbacks[activeEventList[POINTER_ENTER]].length - 1; (callback = handlers.callbacks[activeEventList[POINTER_ENTER]][i]); i--) {
+										// save the pointerId
+										handlers.pointerId = e.pointerId;
+										if (callback(e) === false) {
+											// stop propagating the event if return false
+											handled = true;
+											break;
+										}
+									}
+								}
+							}
+							break;
+						case POINTER_DOWN:
+							// event inside of bounds: trigger the POINTER_DOWN callback
+							if (eventInBounds) {
+								// save the pointerId
+								handlers.pointerId = e.pointerId;
+								// trigger the corresponding callback
+								if (handlers.callbacks[e.type]) {
+									for (i = handlers.callbacks[e.type].length - 1; (callback = handlers.callbacks[e.type][i]); i--) {
+										if (callback(e) === false) {
+											// stop propagating the event if return false
+											handled = true;
+											break;
+										}
+									}
+								}
+							}
+							break;
+						case POINTER_UP:
+							// pointer defined & inside of bounds: trigger the POINTER_UP callback
+							if (handlers.pointerId === e.pointerId && eventInBounds) {
+								// delete the pointerId
+								handlers.pointerId = null;
+								// trigger the corresponding callback
+								if (handlers.callbacks[e.type]) {
+									for (i = handlers.callbacks[e.type].length - 1; (callback = handlers.callbacks[e.type][i]); i--) {
+										if (callback(e) === false) {
+											// stop propagating the event if return false
+											handled = true;
+											break;
+										}
+									}
+								}
+							}
+							break;
+						case POINTER_CANCEL:
+							// pointer defined: trigger the POINTER_CANCEL callback
+							if (handlers.pointerId === e.pointerId) {
+								// delete the pointerId
+								handlers.pointerId = null;
+								// trigger the corresponding callback
+								if (handlers.callbacks[e.type]) {
+									for (i = handlers.callbacks[e.type].length - 1; (callback = handlers.callbacks[e.type][i]); i--) {
+										if (callback(e) === false) {
+											// stop propagating the event if return false
+											handled = true;
+											break;
+										}
+									}
+								}
+							}
+							break;
+					}
+				}
+			}
```

Comment:
Our coding conventions require 4 spaces instead of tab characters. Please convert all tabs to spaces in both files that were modified. That will help align the code properly on github, and make the rest of it easier to review.

Also, we have a file in the project root directory for editor configuration that will help for these kinds of settings: https://github.com/melonjs/melonJS/blob/master/.editorconfig